src/Service/S3/src/S3Client.php

```diff
@@ -651,9 +651,188 @@ public function uploadPart($input): UploadPartOutput
+                return [
+                    'endpoint' => 'https://s3-fips-us-gov-west-1.amazonaws.com',
+                    'signRegion' => 'us-gov-west-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
+                        1 => 's3v4',
+                    ],
+                ];
+            case 's3-external-1':
+                return [
+                    'endpoint' => 'https://s3-external-1.amazonaws.com',
+                    'signRegion' => 'us-east-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
+                        1 => 's3v4',
+                    ],
+                ];
+            case 'sa-east-1':
+                return [
+                    'endpoint' => 'https://s3.sa-east-1.amazonaws.com',
+                    'signRegion' => 'sa-east-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
+                        1 => 's3v4',
+                    ],
+                ];
+            case 'us-east-1':
+                return [
+                    'endpoint' => 'https://s3.us-east-1.amazonaws.com',
+                    'signRegion' => 'us-east-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
+                        1 => 's3v4',
+                    ],
+                ];
+            case 'us-gov-east-1':
+                return [
+                    'endpoint' => 'https://s3.us-gov-east-1.amazonaws.com',
+                    'signRegion' => 'us-gov-east-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
+                        1 => 's3v4',
+                    ],
+                ];
+            case 'us-gov-west-1':
+                return [
+                    'endpoint' => 'https://s3.us-gov-west-1.amazonaws.com',
+                    'signRegion' => 'us-gov-west-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
+                        1 => 's3v4',
+                    ],
+                ];
+            case 'us-iso-east-1':
+                return [
+                    'endpoint' => 'https://s3.us-iso-east-1.c2s.ic.gov',
+                    'signRegion' => 'us-iso-east-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3v4',
+                    ],
+                ];
+            case 'us-west-1':
+                return [
+                    'endpoint' => 'https://s3.us-west-1.amazonaws.com',
+                    'signRegion' => 'us-west-1',
+                    'signService' => 's3',
+                    'signVersions' => [
+                        0 => 's3',
```

Comment:
Do we actually need to cover the non-v4 signature support for some regions ? Only v4 is implemented in the project anyway (which is supported everywhere)

Reply:
`s3v4` is also supported in S3Client. From my last test V4 does not works on S3 endpoints.

I hesitate to filter this list to just keep the best version we knwo. But this means, people won't be able to add their own implementation (by extenting a Client for Instance).

WDYT

Reply:
Well, I know that the s3 client must used the s3-specific signature for things to work (so `s3v4` rather than `v4`). But here, the array also lists `s3`, which is the s3-specific variant of **v2** in the official SDK.

Reply:
Added filter, that returns only the versions supported by the client.

note: `s3` is still returned in the list, because the client currently provides a (wrong) implementation of `s3`. In fact this is a `s3v4`  , but aliased `s3`.

This will be fixed when deprecating the `getSignatureVersion`, which will return `s3v4` in a separate PR.